### DISCUSSION TITLE
sourcecode in testcase read from other file and multi id test

### DISCUSSION THREAD

**Discussion by zoomer-k** [ast-grep/ast-grep#1054] [Q&A]
thank you for producing powerful and useful tool.

Im trying make rule and rule test.

i want to separate rule-test yml and valid or invalid source code
because reuse source code for other test.
(for detect side effect)

other approach is adding multi id in single testcase yml

can i do that?

**Comment by HerringtonDarkholme** [comment]
I would recommend you to use tools to generate YAML files.

**Comment by zoomer-k** [comment]
thank you for your recommned Ill try it

**Reply by zoomer-k** [reply]
https://github.com/ast-grep/ast-grep/issues/1555
